lib/semian.rb

```diff
@@ -216,7 +217,7 @@ def unregister_all_resources
 
   # Retrieves a hash of all registered resources.
   def resources
-    @resources ||= {}
+    @resources ||= LRUCache.new(100)
```

Comment:
The hardcoded limit of 100 seems problematic, but I'm sure you just put it here as a placeholder. 👍  It means that this is _not_ backward compatible. In Shopify, we have way more than 100 resources already—so this would be fairly destructive. I think we can resolve this in two ways without requiring a major version bump for other consumers of Semian outside of Shopify:

* Only destroy resources that have no used bulkheads and no errors in the window
* Provide a configurable limit

(2) is going to mean we'll need to manually count this in every app, which will age poorly. (1) seems hard, but I think might be worth it 🤔 WDYT?

Reply:
I've updated the PR and I think for now we could simply delete resources that are not using bulkheads and the errors are empty. Once we see if it works, we could add a new config variable that could limit the number of resources possible at any given time and if we reach that limit, we delete the oldest one. Adding a limit might not be that useful, I'll do some benchmarking.